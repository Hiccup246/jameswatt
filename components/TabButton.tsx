export default function TabButton({ companyName, activated, clickHandler }: { companyName:string, activated: boolean, clickHandler:Function }) {
  const activatedClasses = activated ? "text-black activated" : "";

  return (
    <button
      aria-label={companyName}
      className={`
        cursor-pointer relative flex w-fit
        h-[var(--tab-height)]
        leading-[var(--tab-height)] px-5 bg-transparent
        whitespace-nowrap z-20 text-black
        mb-[var(--tab-margin-top)] focus-visible:border-none
        max-small:mb-0 max-small:max-w-[var(--tab-width)]
        max-small:grow-0 max-small:shrink-0 max-small:basis-[var(--tab-width)]
        max-small:justify-center
        ${activatedClasses}
      `}
      onClick={(element) => clickHandler(element.currentTarget)}
    >
      {companyName}
    </button>
  );
}