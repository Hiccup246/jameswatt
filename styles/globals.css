@tailwind base;
@tailwind components;
@tailwind utilities;

@layer base {
  :root {
    --slider-items: 7;
    --slider-item-sets: 2;
    --all-slider-items: calc(var(--slider-items) * var(--slider-item-sets));
    --slider-item-width: 200px;
  }

  html,
  body {
    padding: 0;
    margin: 0;
    font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
      "Segoe UI", Roboto, "Helvetica Neue", Arial, "Noto Sans", sans-serif,
      "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol",
      "Noto Color Emoji";
    @apply bg-brown;
  }

  @property --gradient-angle {
    syntax: "<angle>";
    inherits: false;
    initial-value: 1turn;
  }
}

@layer utilities {
  .animation-pause {
    animation-play-state: paused;
  }

  .animation-play {
    animation-play-state: running;
  }
}
