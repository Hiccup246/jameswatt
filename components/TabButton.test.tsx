import { expect, test } from "@jest/globals";
import TabButton from "./TabButton";
import renderer from "react-test-renderer";

test("renders the TabButton correctly", () => {
  const tree = renderer
    .create(
      <TabButton
        companyName="Company One"
        activated={true}
        clickHandler={() => {}}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
