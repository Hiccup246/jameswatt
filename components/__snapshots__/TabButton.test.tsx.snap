// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TabButton correctly 1`] = `
<button
  aria-label="Company One"
  className="
        bg-transparent relative z-20 mb-[var(--tab-margin-top)]
        flex
        h-[var(--tab-height)] w-fit cursor-pointer
        whitespace-nowrap px-5 leading-[var(--tab-height)]
        text-black focus-visible:border-none
        max-small:mb-0 max-small:max-w-[var(--tab-width)]
        max-small:shrink-0 max-small:grow-0 max-small:basis-[var(--tab-width)]
        max-small:justify-center
        text-black activated
      "
  onClick={[Function]}
>
  Company One
</button>
`;
